alacritty_terminal/src/term/mod.rs

```diff
@@ -1309,14 +1316,19 @@ impl Term {
         &mut self.clipboard
     }
 
-    pub fn urls(&self) -> Vec<Url> {
+    pub fn update_urls(&mut self) {
         let display_offset = self.grid.display_offset();
         let num_cols = self.grid.num_cols().0;
         let last_col = Column(num_cols - 1);
         let last_line = self.grid.num_lines() - 1;
 
-        let grid_end_point = Point::new(0, last_col);
-        let mut iter = self.grid.iter_from(grid_end_point);
+        let max_above = Linear::new(Column(num_cols), Column(0), last_line + Line(display_offset))
+            + Linear(MAX_URL_LENGTH);
+
+        let start_line = display_offset.saturating_sub(MAX_URL_LENGTH / num_cols);
+        let start_col = if start_line == 0 { last_col } else { Column(MAX_URL_LENGTH % num_cols) };
```

Comment:
Is it correct that this might go slightly beyond `MAX_URL_LENGTH` when on the last line, since we always go to the last column here instead of making sure that the URL wouldn't have ended before it?

It seems like always returning `last_col` here is to make the codepath simpler where `saturating_sub` would usually make the line go negative, is that correct?

Reply:
I can't see any situations when this will go beyond `MAX_URL_LENGTH`; am I missing some edge case?

`start_line` will never be negative, because it's a `usize` and `saturating_sub` clamps it to zero. The reason behind the `if` statement there is to ensure that when there are fewer characters past the viewport than `MAX_URL_LENGTH` that all of them are checked.

Reply:
This case is not only triggered when the length goes beyond the bottom-most line, but it's also triggered when it stops right in it. Then the columns will still get artificially extended to the last column, even though it would not be necessary.

This is because like this there's no way to differentiate between a `0` starting line which was truncated and one which was not.

At least that's my understanding. However that might not necessarily be a problem since it shouldn't make a massive difference to the URL length, however we should be aware that this basically makes it possible in theory to have infinite length URLs.

Reply:
Alright, thanks for explaining it - does 
63aa56d solve this?